ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,64 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner task.
+
+
+## Decision
+
+### How would the sensitive information be persisted and made available to the PipelineRun ?
+
+1. The UI would take in sensitive information as form input and upload it to SPI using the user's authentication token. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret .
+
+2. Subsequently, the UI would create an `SPIAccessTokenBinding` CR with the appropriate label in the `.spec.secret.labels` to make the `Secrets` available to the build service or the integration service.
```

Comment:
Does the `SPIAccessTokenBinding` expire?

What happens when future build pipelineruns are started by PaC after it has expired? Should something be responsible for re-generating it just-in-time?

Reply:
The CR has an expiration setting. There is a default but you can set to `-1` for no expiration.

Reply:
Today I learned!